bindings/python/pymongoarrow/api.py

```diff
@@ -279,17 +292,23 @@ def write(collection, tabular):
     :Returns:
       An instance of :class:`result.ArrowWriteResult`.
     """
-
+    tab_size = len(tabular)
     if isinstance(tabular, Table):
         _validate_schema(tabular.schema.types)
     elif isinstance(tabular, DataFrame):
         _validate_schema(ArrowSchema.from_pandas(tabular).types)
+    elif (
+        isinstance(tabular, dict)
+        and len(tabular.values()) >= 1
+        and isinstance(list(tabular.values())[0], ndarray)
```

Comment:
Do we need to check `tabular.values()` in this if statement? What happens if `tabular` is a dict with different format?

Reply:
The current code is still problematic, for example what happens when you call:
```
write({'foo': 1})
```